qpeerset/qpeerset.go

```diff
@@ -136,12 +140,13 @@ func (qp *QueryPeerset) GetWaitingPeers() (result []peer.ID) {
 	return
 }
 
-// GetClosestNotUnreachable returns the closest to the key peers, which are not in state PeerUnreachable.
+// GetClosestNotUnreachableOrRejected returns the closest to the key peers, which are not in state PeerUnreachable or in any of the "Rejected" states.
 // It returns count peers or less, if fewer peers meet the condition.
-func (qp *QueryPeerset) GetClosestNotUnreachable(count int) (result []peer.ID) {
+// The returned peers are sorted in ascending order by their distance to the key.
+func (qp *QueryPeerset) GetClosestNotUnreachableOrRejected(count int) (result []peer.ID) {
```

Comment:
WDYT about replacing all of these `GetXYZ` functions with `GetClosestWithStates(count int, states... PeerState)`. Seems like it'd make the code a bit easier to work with as we add new states. The downside is that it might be useful for this to be a little bit miserable when we change the state machine to make sure we do so carefully.

Reply:
After trying this out, it does look good.  Have fixed this.